src/main/java/com/serhiiklymchuk/gmailpet/controller/UserController.java

```diff
@@ -0,0 +1,48 @@
+package com.serhiiklymchuk.gmailpet.controller;
+
+import com.serhiiklymchuk.gmailpet.dto.UserDto;
+import com.serhiiklymchuk.gmailpet.service.impl.UserServiceImpl;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import javax.validation.Valid;
+
+@Controller
+@RequestMapping("/user")
+public class UserController {
+
+    private final UserServiceImpl userService;
+
+    public UserController(UserServiceImpl userService) {
+        this.userService = userService;
+    }
+
+    @GetMapping("/home")
+    public String getHome(){
+        return "user/home";
+    }
+
+    @GetMapping("/register")
+    public String getRegistrationPage(){
+        return "user/register";
+    }
+
+    @PostMapping("/register")
+    public String createUser(@Valid UserDto userDto, BindingResult bindingResult, RedirectAttributes attr, Model model){
+
+        if(bindingResult.hasErrors()){
+            model.addAttribute("errors", bindingResult.getAllErrors());
+            return "user/register";
+        }
+
+        userService.createUser(userDto);
+        attr.addFlashAttribute("user", userDto);
```

Comment:
`redirect: /home`

Reply:
It doesn't work this way(redirects to `/login`)
using `redirect: home` redirects to user/home after registration as it is supposed to

Reply:
It's because the path in your case is `/user/home` and it's the reason why you should always you absolute path (starting with `/`) when redirecting, because, in another case, you will be redirecting relatively to the path, where user was before, so it's buggy prone